tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/tagless/algebras/ui/Presentation.kt

```diff
@@ -33,46 +36,44 @@ interface SuperHeroDetailView : SuperHeroesView {
 
 }
 
-@typeclass
-interface Presentation<F> : TC {
-
-    fun navigation(): Navigation<F>
-
-    fun heroesService(): HeroesUseCases<F>
-
-    fun ME(): MonadError<F, Throwable>
-
-    fun onHeroListItemClick(ctx: Context, heroId: String): HK<F, Unit> =
-            navigation().goToHeroDetailsPage(ctx, heroId)
-
-    fun displayErrors(view: SuperHeroesView, t: Throwable): HK<F, Unit> =
-            ME().pure(when (CharacterError.fromThrowable(t)) {
-                is CharacterError.NotFoundError -> view.showNotFoundError()
-                is CharacterError.UnknownServerError -> view.showGenericError()
-                is CharacterError.AuthenticationError -> view.showAuthenticationError()
-            })
+class Presentation<F> @Inject constructor(
```

Comment:
Same. I am a bit between 2 worlds with this approach. I see that is a pretty big step forward towards compiletime validation and instance injection, but at the same time is really hard for users starting to learn the paradigm to assotiate with the Typeclass concept. I can imagine them having big trouble on inferring that by themselves by looking at how the code looks like when declaring the typeclasses like this.